pkg/storegateway/store_gateway_client.go

```diff
@@ -47,32 +47,32 @@ func dialStoreGatewayClient(clientCfg grpcclient.Config, addr string, requestDur
 		return nil, errors.Wrapf(err, "failed to dial store-gateway %s", addr)
 	}
 
-	return &storeGatewayClient{
+	return &ClientImpl{
 		StoreGatewayClient: storegatewaypb.NewStoreGatewayClient(conn),
 		HealthClient:       grpc_health_v1.NewHealthClient(conn),
 		conn:               conn,
 	}, nil
 }
 
-type storeGatewayClient struct {
+type ClientImpl struct {
```

Comment:
If this is implementing an interface, we could refer to the interface everywhere and make the actual struct unexported.

Reply:
Will be fixed in a separate PR